### PR TITLE
Enhance API docs of portal.translate to show that the domain is optio…

### PR DESCRIPTION
**Pull request by thet**
…nal in some cases.

If anyone can point me to some documentation on the api doc syntax, I'd be happy. Didn't found anything than docs on how to use sphinx autodoc on an API level and so forth.

### CODE REVIEW

File: src/plone/api/portal.py [hunk 1]
```diff
@@ -407,8 +407,9 @@ def translate(msgid, domain="plone", lang=None):
     Default to current negotiated language if no target language specified.
 
     :param msgid: [required] message to translate
-    :type msgid: string
-    :param domain: i18n domain to use
+    :type msgid: string | zope.i18nmessageid.Message
+    :param domain: i18n domain to use. When `msgid` is a instance of `Message`
+                   the Message' domain is used.
```
Comment (thet) [inline src/plone/api/portal.py:412]: Is the intend on this line right or will it fail any docstring formatters?

File: src/plone/api/portal.py [hunk 2]
```diff
@@ -407,8 +407,9 @@ def translate(msgid, domain="plone", lang=None):
     Default to current negotiated language if no target language specified.
 
     :param msgid: [required] message to translate
-    :type msgid: string
-    :param domain: i18n domain to use
+    :type msgid: string | zope.i18nmessageid.Message
+    :param domain: i18n domain to use. When `msgid` is a instance of `Message`
```
Comment (stevepiercy) [inline src/plone/api/portal.py:411]: docstrings need to follow reST syntax, not MyST syntax. I know, annoying, isn't it?

```suggestion
    :param domain: i18n domain to use. When ``msgid`` is an instance of ``Message``,
```

File: src/plone/api/portal.py (hunk 1 above)
Comment (stevepiercy) [inline src/plone/api/portal.py:412]:
```suggestion
                   then the ``Message``'s domain is used.
```

File: src/plone/api/portal.py [hunk 3]
```diff
@@ -407,8 +407,9 @@ def translate(msgid, domain="plone", lang=None):
     Default to current negotiated language if no target language specified.
 
     :param msgid: [required] message to translate
-    :type msgid: string
-    :param domain: i18n domain to use
+    :type msgid: string | zope.i18nmessageid.Message
```
[the remark this answers is not in the document]
Comment (stevepiercy) [reply src/plone/api/portal.py:410]: I think it is the following, based on this example: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#python-signatures

```suggestion
    :type msgid: string
    :type msgid: zope.i18nmessageid.Message
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
I don't know for sure, but let's try these suggestions. We'll figure it out.

**Comment by thet** [comment]
@stevepiercy tnx for the review and sorry for my late reply.
I've updated this PR with your suggestions.

File: news/510.documentation [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Enhance API docs of portal.translate to show that the domain is optional in some cases.
```
Comment (stevepiercy) [inline news/510.documentation:1]: Sprinkle some MyST in there, standardize author.

```suggestion
Enhance API docs of `portal.translate` to show that the domain is optional in some cases. @thet
```

File: news/510.documentation [hunk 5]
```diff
@@ -0,0 +1,2 @@
...
+[thet]
```
Comment (stevepiercy) [inline news/510.documentation:2]:
```suggestion
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Looks good, just a little tweak for the change log, and it is good to go. Thank you!

**Comment by thet** [comment]
Thank you! I applied your suggestions. I'm merging.

**Pull request merged**